packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -14,6 +15,52 @@ import 'debug.dart';
 import 'image_provider.dart' as image_provider;
 import 'image_stream.dart';
 
+class _DownloadRequest {
+  _DownloadRequest(this.sendPort, this.uri, this.headers);
+
+  SendPort sendPort;
+  Uri uri;
+  Map<String, String> headers;
+}
+
+class _HttpClientIsolateParameters {
+  _HttpClientIsolateParameters(this.sendPort, this.chunkEvents);
+
+  SendPort sendPort;
+  StreamController<ImageChunkEvent> chunkEvents;
+}
+
+void _handleHttpClientGet(_HttpClientIsolateParameters params) {
+  final HttpClient _httpClient = HttpClient();
+  final RawReceivePort receivePort = RawReceivePort((_DownloadRequest downloadRequest) async {
+    try {
+      final HttpClientRequest request = await _httpClient.getUrl(downloadRequest.uri);
+      downloadRequest.headers?.forEach((String name, String value) {
+        request.headers.add(name, value);
+      });
+      final HttpClientResponse response = await request.close();
+      if (response.statusCode != HttpStatus.ok) {
+        throw Exception(
+            'HTTP request failed, statusCode: ${response
+                ?.statusCode}, ${downloadRequest.uri}');
+      }
+      final TransferableTypedData transferable = await consolidateHttpClientResponseBytes(
+          response,
+          onBytesReceived: (int cumulative, int total) {
+            params.chunkEvents.add(ImageChunkEvent(
+              cumulativeBytesLoaded: cumulative,
+              expectedTotalBytes: total,
+            ));
+          });
+      downloadRequest.sendPort.send(transferable);
+    } catch (error) {
+      downloadRequest.sendPort.send(error.toString());
```

Comment:
Would we be losing the stacktrace when there's an error? Can we print it to stderr?

Also, I imagine we should be robust w.r.t. network errors. A network error is not necessarily critical and worth logging. We should somehow communicate this to the main isolate for it to decide what to do (try again, log, try something else).